packages/grid/x-data-grid-pro/src/tests/columns.DataGridPro.test.tsx

```diff
@@ -397,17 +397,4 @@ describe('<DataGridPro /> - Columns', () => {
       expect(gridColumnFieldsSelector(apiRef)).to.deep.equal(['__check__', 'brand', 'id']);
     });
   });
-
-  it('should warn if unsupported column type is used', () => {
-    const singleColumns = [{ field: 'brand', type: 'str' }];
-
-    expect(() => {
-      render(<Test columns={singleColumns} />);
-    }).toWarnDev(
-      [
-        `MUI: The column type "str" you are using is not supported.`,
-        `Column type "string" is being used instead.`,
-      ].join('\n'),
-    );
```

Comment:
I'm not sure to follow the rationale. The change in the tests looks like a step back. `type: 'str'` looks wrong. Or maybe it's because this test removal unlocks a new test case but we didn't add it?

Reply:
I think now that the warning is useless since we removed the `columnTypes` prop.

Reply:
Alright, I moved my comment to https://github.com/mui/mui-x/issues/4769#issuecomment-1642592703.